components/datetime/src/pattern/mod.rs

```diff
@@ -281,3 +288,113 @@ impl Serialize for Pattern {
         }
     }
 }
+
+/// Used to represent either H11/H12, or H23/H24. Skeletons only store these
+/// hour cycles as H12 or H24.
+#[derive(Debug, PartialEq, Clone, Copy)]
+#[cfg_attr(
+    feature = "provider_serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
+pub enum FlexibleHourCycle {
+    /// Can either be fields::Hour::H11 or fields::Hour::H12
+    H11H12,
+    /// Can either be fields::Hour::H23 or fields::Hour::H24
+    H23H24,
+}
+
+/// Default is required for serialization, arbitrarily pick one.
+impl Default for FlexibleHourCycle {
+    fn default() -> Self {
+        FlexibleHourCycle::H11H12
+    }
+}
+
+impl FlexibleHourCycle {
```

Comment:
Nit: This file is getting too big.  Please refactor this into its own file.  As a rule of thumb, I like to keep *lib.rs* and *mod.rs* empty except for imports.

Reply:
Done.